pkg/policy/policy.go

```diff
@@ -77,47 +83,72 @@ func NewClient(cacheDir string, quiet bool, checkBundleRepo string, opts ...Opti
 		opt(o)
 	}
 
-	if checkBundleRepo == "" {
-		checkBundleRepo = fmt.Sprintf("%s:%d", BundleRepository, BundleVersion)
+	if len(checkBundleRepos) == 0 {
+		checkBundleRepos = BundleRepositories
 	}
 
 	return &Client{
-		options:         o,
-		policyDir:       filepath.Join(cacheDir, "policy"),
-		checkBundleRepo: checkBundleRepo,
-		quiet:           quiet,
+		options:          o,
+		policyDir:        filepath.Join(cacheDir, "policy"),
+		checkBundleRepos: checkBundleRepos,
+		quiet:            quiet,
 	}, nil
 }
 
-func (c *Client) populateOCIArtifact(ctx context.Context, registryOpts types.RegistryOptions) {
+func (c *Client) populateOCIArtifact(ctx context.Context, repo string, registryOpts types.RegistryOptions) {
 	if c.artifact == nil {
-		log.DebugContext(ctx, "Loading check bundle", log.String("repository", c.checkBundleRepo))
-		c.artifact = oci.NewArtifact(c.checkBundleRepo, registryOpts)
+		if repo == "" {
+			repo = c.checkBundleRepos[0]
+		}
+		log.DebugContext(ctx, "Loading check bundle", log.String("repo", repo))
+		c.artifact = oci.NewArtifact(repo, registryOpts)
 	}
 }
 
-// DownloadBuiltinChecks download default policies from GitHub Pages
+// DownloadBuiltinChecks download default policies from OCI registry
 func (c *Client) DownloadBuiltinChecks(ctx context.Context, registryOpts types.RegistryOptions) error {
-	c.populateOCIArtifact(ctx, registryOpts)
-
-	dst := c.contentDir()
-	if err := c.artifact.Download(ctx, dst, oci.DownloadOption{
-		MediaType: policyMediaType,
-		Quiet:     c.quiet,
-	},
-	); err != nil {
-		return xerrors.Errorf("download error: %w", err)
-	}
-
-	digest, err := c.artifact.Digest(ctx)
-	if err != nil {
-		return xerrors.Errorf("digest error: %w", err)
-	}
-	log.DebugContext(ctx, "Digest of the built-in checks", log.String("digest", digest))
-
-	// Update metadata.json with the new digest and the current date
-	if err = c.updateMetadata(digest, c.clock.Now()); err != nil {
-		return xerrors.Errorf("unable to update the check metadata: %w", err)
+	oldart := c.artifact
+
+	for i, repo := range c.checkBundleRepos {
+		c.populateOCIArtifact(ctx, repo, registryOpts)
```

Comment:
This will not work because the artifact is only initialized if it is missing, so the old one will be used. https://github.com/aquasecurity/trivy/blob/main/pkg/policy/policy.go#L92